application/controllers/handler.js

```diff
@@ -274,7 +281,7 @@ module.exports = {
         updateQuery = {
           $set: {
             email:       email.toLowerCase(),
-            username:    util.parseAPIParameter(req.payload.username),
+            username:    util.parseAPIParameter(req.payload.username).replace(/\s/g,''),
```

Comment:
Redundant to remove whitspaces because they are already handled at the route itself